### DISCUSSION TITLE
Split build file into smaller pieces

### DISCUSSION THREAD

**Discussion by Naaaaiix** [maxi-blocks/maxi-blocks#4766] [Q&A]
One of the comments received about Maxi is related with the metrics we're scoring through this [comparison report](https://wphive.com/compare/plugins/maxi-blocks/vs/kadence-blocks/) where, in the section `Page Speed`, we can see a huge increase of 4sec when opening the editor due to MaxiBlocks installation:

<img width="858" alt="Captura de Pantalla 2023-04-25 a las 11 27 29" src="https://user-images.githubusercontent.com/50477222/234234711-7b1ce17f-4020-4e80-8aa0-46e514dc5ccb.png">

To be sure, been checking the timeout scores in different scenarios, here some metrics:
1. WP without Maxi => 4.4 MB transferred || Load: 1.02 s
2. WP with Maxi v.1.0.0-beta-2 => 6.7 MB transferred || Load: 2.61 s
3. WP with Maxi v1.0.1 => 6.8 MB/6.9 MB transferred || Load: 2.36 s

I used beta-2 as is the one with the speed optimizations we used as reference. The basic difference is in the MaxiBlocks main javascript file, that is 2.2mb. There's no way to modify its async or deferred behavior, as its controlled by WP to be available when Gutenberg requires the register of the blocks. This file delays 0.9s the load of the editor, so I guess the rest of the difference is coming from styles. 
When we see the waterfall of the Gutenberg editor loading we can appreciate it loads all the JS files that will register blocks in an asynchronous way and once all have been loaded, it start the registering of the blocks. What other plugins does is to split the JS file into smaller ones and has a JS file for each block, which reduces the amount of time as the smaller parts are loaded asynchronous; Maxi, as it's a unique file of 2.2mb increases the time.

One solution I find to fix this situation is to split Maxi file into smaller blocks files 👍 (I'll be expanding this soon)

**Comment by Naaaaiix** [comment]
This can be really helpful => https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#automatic-block-json-detection-and-the-source-code-directory

**Reply by elzadj** [reply]
Exactly what we need, but requires us to add block registration to block.json
https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/

Welp, it's a great optimization step, so...

**Reply by Naaaaiix** [reply]
Yes, great one. Been working on a script that will create the block.json automatically to each one of our blocks, so that part of the job is mostly done. The good part of having the block.json created is that it will make much easier to register blocks on our PHP side, and with time we'll be able to do things like migrate the blocks from the PHP side (much much easier) or load the frontend styles from PHP side 👍 

I'm excited with this optimization 💪 😸

**Reply by kyrapieterse** [reply]
I never thought I would see excited and optimization in the same sentence 🧐
